src/EditorFeatures/Core/Shared/Utilities/AbstractClassificationTypeMap.cs

```diff
@@ -4,58 +4,87 @@
 
 using System;
 using System.Collections.Generic;
-using System.Text;
 using Microsoft.CodeAnalysis.Classification;
 using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.VisualStudio.Text.Classification;
 using Roslyn.Utilities;
 using ReferenceEqualityComparer = Roslyn.Utilities.ReferenceEqualityComparer;
 
-namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+
+/// <summary>
+/// Note: we very intentionally layer things in the following fashion (from lowest to highest priority):
+/// <list type="number">
+/// <item>
+/// Roslyn Syntactic Classifications.  These go into the "Editor Lexical" bucket, lower than everything else.
+/// </item>
+/// <item>
+/// Roslyn Semantic Classifications.  These go into the "Editor Syntactic" bucket.  Effectively *almost* always beating
+/// out the classifications produced by Roslyn's syntactic classifier.
+/// </item>
+/// <item>
+/// Comments.  These go into the highest bucket "Editor Semantic".  This ensures that comments override *everything*,
+/// causing classification to instantly 'snap' to the commented-out classification when the user comments code.  Without
+/// this, our stale semantic classifications could show up *above* the commented code *up until* the point that semantic
+/// classification ran again.
+/// </item>
+/// </list>
+/// </summary>
```

Comment:
📝 This feels like a very ugly solution to the problem. I believe a better approach would be to have the semantic classifier handle text changes and immediately remove classifications from known comments even before it runs the next delayed semantic classification pass.

Reply:
That doesn't feel clean to me.  It would require some sort of path for taggers to pre-run, do pre-computation and mutate results prior to then doing the actual pass.

Reply:
It wouldn't necessarily need that. The semantic classification could be presented to the editor through an aggregation/filtering layer that takes both taggers as inputs, and emits an output which contains all the semantic tags except for the ones hidden by comments. This would allow for a cleaner mapping of taggers to editor classification layers, and also make it clear where the intentional layer inversion is occurring for latency purposes. It would also make it easier to account for similar scenarios should they arise in the future (e.g. disabled code from preprocessor directives).

Reply:
i think the point of hte ClassificationLayer is to do exactly that.  it's just that the aggregation/filtering layer is handled by them, instead of us.  

we could attempt to move something lower down to us.  i'm not opposed to that.  i just don't feel strongly that it would help that much to own this ourselves.